packages/plugins/rudder_plugin_interface/lib/src/models/rudder_config.dart

```diff
@@ -152,51 +149,98 @@ class RudderConfig {
-        webConfig.cookieConsentManagers?.forEach((key, value) {
-          cookieConsent[key] = {"enabled": true};
-        });
-        _webConfigMap["cookieConsentManager"] = cookieConsent;
+      // consent manager
+      if (webConfig.consentManagement != null) {
+        _webConfigMap["consentManagement"] =
+            webConfig.consentManagement?.toMap();
       }
-      if (Utils.isValidUrl(webConfig.destSDKBaseURL)) {
-        _webConfigMap["destSDKBaseURL"] = webConfig.destSDKBaseURL;
-      } else {
-        RudderLogger.logWarn("Dest SDK Base Url is not valid, using default");
-        _webConfigMap["destSDKBaseUrl"] =
-            Constants.DEFAULT_DESTINATION_SDK_BASE_URL;
+      if (webConfig.sessions != null) {
+        _webConfigMap["sessions"] = webConfig.sessions?.toMap();
       }
-      sessionOpts['autoTrack'] = webConfig.autoSessionTracking;
-      if (webConfig.sessionTimeoutInMillis < 0) {
-        RudderLogger.logError("invalid sessionTimeoutInMillis. Set to default");
-        sessionOpts['timeout'] = Constants.DEFAULT_SESSION_TIMEOUT_WEB;
-      } else {
-        sessionOpts['timeout'] = webConfig.sessionTimeoutInMillis;
+      if (webConfig.destSDKBaseURL != null) {
+        if (Utils.isValidUrl(webConfig.destSDKBaseURL as String)) {
+          _webConfigMap["destSDKBaseURL"] = webConfig.destSDKBaseURL;
+        } else {
+          RudderLogger.logWarn("Dest SDK Base Url is not valid, using default");
+        }
+      }
+      if (webConfig.pluginsSDKBaseURL != null) {
+        if (Utils.isValidUrl(webConfig.pluginsSDKBaseURL as String)) {
+          _webConfigMap["pluginsSDKBaseURL"] = webConfig.pluginsSDKBaseURL;
+        } else {
+          RudderLogger.logWarn(
+              "Plugin SDK Base Url is not valid, using default");
+        }
+      }
+      if (webConfig.storage != null) {
+        _webConfigMap["storage"] = webConfig.storage?.toMap();
+      }
+      if (webConfig.destinationsQueueOpts != null) {
+        _webConfigMap["destinationsQueueOpts"] =
+            webConfig.destinationsQueueOpts?.toMap();
+      }
+      if (webConfig.anonymousIdOptions != null) {
+        _webConfigMap["anonymousIdOptions"] =
+            webConfig.anonymousIdOptions?.toMap();
+      }
+      if (webConfig.onLoaded != null) {
+        _webConfigMap["onLoaded"] = webConfig.onLoaded;
+      }
+      if (webConfig.sendAdblockPageOptions != null) {
+        _webConfigMap["sendAdblockPageOptions"] =
+            webConfig.sendAdblockPageOptions?.toMap();
+      }
+      if (webConfig.plugins != null) {
+        _webConfigMap["plugins"] = webConfig.plugins;
+      }
+      if (webConfig.polyfillURL != null) {
+        if (Utils.isValidUrl(webConfig.polyfillURL as String)) {
+          _webConfigMap["polyfillURL"] = webConfig.polyfillURL;
+        } else {
+          RudderLogger.logWarn(
+              "Provided Polyfill Url is not valid, using default");
+        }
+      }
```

Comment:
Let's not do this validation here. The JS SDK will anyway repeat it.

Reply:
Done